src/subtype.c

```diff
@@ -355,25 +357,24 @@ static int obviously_disjoint(jl_value_t *a, jl_value_t *b, int specificity)
             np = jl_nparams(ad);
         }
         size_t i;
-        for(i=0; i < np; i++) {
-            jl_value_t *ai = jl_tparam(ad,i);
-            jl_value_t *bi = jl_tparam(bd,i);
+        for (i = 0; i < np; i++) {
+            jl_value_t *ai = jl_tparam(ad, i);
+            jl_value_t *bi = jl_tparam(bd, i);
             if (jl_is_typevar(ai) || jl_is_typevar(bi))
-                continue;
+                continue; // Union{} is always in this intersection
```

Comment:
I don't think this is true, e.g. if there is a non-`Union{}` lower bound. We just skip these because vars are difficult to handle.

Reply:
```suggestion
                continue; // it's possible that Union{} is in this intersection
```